graphql/end_to_end_test.go

```diff
@@ -830,3 +832,121 @@ func TestConcurrencyLimiterDeadlock(t *testing.T) {
 	wg.Wait()
 	defer rerunner.Stop()
 }
+
+func TestSkipDirectives(t *testing.T) {
+	schema := schemabuilder.NewSchema()
+	query := schema.Query()
+	query.FieldFunc("value", func() string { return "s" })
+	builtSchema := schema.MustBuild()
+
+	execute := func(queryString string, vars map[string]interface{}) (interface{}, error) {
+		q := graphql.MustParse(queryString, vars)
+
+		if err := graphql.PrepareQuery(builtSchema.Query, q.SelectionSet); err != nil {
+			return nil, err
+		}
+
+		e := graphql.Executor{}
+		return e.Execute(context.Background(), builtSchema.Query, nil, q)
+	}
+
+	// Variable skip
+	result, err := execute(`
+		query x {
+			value @skip(if: $var)
+		}`, map[string]interface{}{"var": true})
+	if err != nil {
```

Comment:
out of curiosity - is there a reason for preferring the use of `if err != nil` + `t.Errorf` over `assert.NoError` here?

Reply:
not really. this code was originally written in jan when we didn't use testify yet.